php/WP_CLI/Fetchers/User.php

```diff
@@ -24,8 +25,24 @@ class User extends Base {
 	 */
 	public function get( $arg ) {
 
+		if ( getenv( 'WP_CLI_FORCE_USER_LOGIN' ) ) {
+			$this->msg = "Invalid user login: '%s'";
+			return get_user_by( 'login', $arg );
+		}
+
 		if ( is_numeric( $arg ) ) {
-			$user = get_user_by( 'id', $arg );
+			$check = get_user_by( 'login', $arg );
+			$user  = get_user_by( 'id', $arg );
+			if ( $check && $user ) {
+				WP_CLI::warning(
+					sprintf(
+						'Ambiguous user match (ID=%d and user_login=%d). Defaulting to ID. Force user_login with WP_CLI_FORCE_USER_LOGIN=1.',
+						$arg,
+						$arg,
+						$arg
```

Comment:
I think the message could be phrased a bit clearer. Something like this maybe?
```suggestion
						'Ambiguous user match detected (both ID and user_login exist for identifier "%d"). WP-CLI will default to the ID, but you can make it use the user_login instead with WP_CLI_FORCE_USER_LOGIN=1.',
						$arg,
						$arg
```
(test will need to be adapted as well)

Reply:
Updated with a61e9753